scale_info/transformTys.ts

```diff
@@ -0,0 +1,108 @@
+import * as $ from "../deps/scale.ts"
+import { Codec } from "../deps/scale.ts"
+import { getOrInit } from "../util/mod.ts"
+import { normalizeIdent } from "../util/normalize.ts"
+import { overrides } from "./overrides/mod.ts"
+import { Ty } from "./raw/Ty.ts"
+
+/**
+ * All derived codecs for ZSTs will use this exact codec,
+ * so `derivedCodec === $null` is true iff the type is a ZST.
+ */
+export const $null = $.constant(null)
+
+export function transformTys(tys: Ty[]): [Codec<any>[], Record<string, Codec<any>>] {
+  const memo = new Map<number, Codec<any>>()
+  const paths: Record<string, Codec<any>> = {}
```

Comment:
- Assuming you're leaving populating this record for later?
- Where will be utilize `paths`?

Reply:
Later in this PR, yes.

`paths` will be a e.g.
```ts
{
  "sp_runtime.multiaddress.MultiAddress": $.taggedUnion(...),
  ...
}
```
and we'll primarily use this to populate the `types/` directory in the codegen.